spark/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala

```diff
@@ -105,19 +107,24 @@ trait SnapshotManagement { self: DeltaLog =>
    * @return Some array of files found (possibly empty, if no usable commit files are present), or
    *         None if the listing returned no files at all.
    */
-  protected final def listDeltaAndCheckpointFiles(
+  protected final def listDeltaCompactedDeltaAndCheckpointFiles(
       startVersion: Long,
-      versionToLoad: Option[Long]): Option[Array[FileStatus]] =
+      versionToLoad: Option[Long],
+      includeMinorCompactions: Boolean): Option[Array[FileStatus]] =
     recordDeltaOperation(self, "delta.deltaLog.listDeltaAndCheckpointFiles") {
       listFromOrNone(startVersion).map { _
-        // Pick up all checkpoint and delta files
-        .filter { file => isDeltaCommitOrCheckpointFile(file.getPath) }
-        // Checkpoint files of 0 size are invalid but Spark will ignore them silently when reading
-        // such files, hence we drop them so that we never pick up such checkpoints.
-        .filterNot { file => isCheckpointFile(file) && file.getLen == 0 }
+        .collect {
+          case DeltaFile(f, fileVersion) =>
+            (f, FileType.DELTA, fileVersion)
+          case CompactedDeltaFile(f, startVersion, endVersion)
+            if includeMinorCompactions && versionToLoad.forall(endVersion <= _) =>
+            (f, FileType.COMPACTED_DELTA, startVersion)
+          case CheckpointFile(f, fileVersion) if f.getLen > 0 =>
+            (f, FileType.CHECKPOINT, fileVersion)
```

Comment:
I don't see any use of the middle element of this triple? The version is used for `takeWhile` and the file status is what we actually return... but where is file type used?